app/dao/jobs_dao.py

```diff
@@ -30,10 +30,10 @@
 from app.utils import midnight_n_days_ago
 
 
-def dao_get_notification_outcomes_for_job(service_id, job_id):
+def dao_get_notification_outcomes_for_job(job_id):
     notification_statuses = (
```

Comment:
idly thinking, we could be smarter and instead of just saying "look in notifications, if it's empty look in ft_notification_status", we could say `if job.created_at > last_midnight` before needing to query

actually never mind i just realised that this would mean the stats wouldn't be guaranteed up to date for jobs from yesterday with things still in sending, which is presumably why we did it this way

Reply:
I am strongly considering another change where we detect jobs whose statuses are all "final", and aggressively redis-cache the result.